src/routes/AppRoutes.tsx

```diff
@@ -13,44 +13,54 @@ import { PrivateRoute } from './PrivateRoute';
 import { authRoutes } from './routesList';
 
 export const AppRoutes = () => {
-  const dispatch = useAppDispatch();
   const token = sessionStorage.getItem('accessToken');
+
+  const [loaded, setLoaded] = useState(!token);
+  const dispatch = useAppDispatch();
+
   const isAuthorized = auth.useAuthorized();
 
   useEffect(() => {
-    if (!isAuthorized && token) {
-      dispatch(auth.thunk.signInWithToken({ token }));
-    }
+    const signInWithToken = async () => {
+      if (!isAuthorized && token) {
+        await dispatch(auth.thunk.signInWithToken({ token }));
+        setLoaded(true);
```

Comment:
because when we reload the dashboard, /auth is loaded first, and then it redirects to /dashboard 
I guess, because at that moment isAuthorized = false and we have to wait for signInWithToken to complete

Reply:
We can use meta from Redux store, to make preloading functionality. And maybe we also need global preloader?